proposals/0001-gep.md

```diff
@@ -0,0 +1,229 @@
+- R2: Config format must allow comments
+- R3: Config format must be `git diff`-able and versioned
+- R4: Config must be easy to understand and reason about
+- R5: Config must allow to include sensitive values from environment variables
+- R6: Config may allow to include sensitive values from other plain files
+- R7: Config should be easy to change and operate on like "hitless" reloads 
+
+## Design
+
+There are basically a few sources of configurations possible:
+- via Glide CLI params & args
+- via environment variables
+- via a separate config file
+- via a combination of CLI, environment variables, and/or config file
+
+Setting complex nested configs via CLI or environment variables is usually a tedious error-prone process.
+So Glide will try to leverage the file as config approach as long as possible.
+
+### The Config Source
+
+Glide config is represented as a YAML file (R1, R2, R3). 
+YAML is a human-friendly version of JSON that supports comments and some useful magic like (including environment variables).
+Go has good support for the YAML format.
+
+### The Config Structure
+
+The config structure uses Glide's internal logical structure that we are going to communicate in Glide docs (R4).
+So learning more about how Glide works should give enough context to understand the config. 
+The opposite should be true as well, so knowing how the config looks will give some understanding about Glide's overall design.
+This is a route that the OTEL Collector config takes.
+
+Glide should allow you to pull values of configs from environment variables (see the config example below).
+This helps to keep sensitive information (R5) out of the config file that avoids leakages.
+
+### Config Ops
+
+Glide should be able to automatically detect changes to the config (especially, our domain-related configs like routing) and
+reload its internal representation together with the related state (R7). 
+In Kubernetes, configs are going to be represented as a configmap mounted into Glide pods. 
+On configmap change, the mounted file changes as well. So we need to watch for the file changes there. 
+Outside of Kubernetes, we may support config reloading on SIGHUP signals.
+
+### Config Example
+
+The most minimalistic possible config setup would look like this:
+
+```yaml
+routes:
+  language:
+    - name: openai-pool
+      providers:
+        - name: openai-boring
+          provider: openai
+          model: gpt-3.5-turbo
+          api_key: ${env:OPENAI_API_KEY}
+          default_params:
+            temperature: 0
+```
+
+Here is a rich config example to get better sense of the config structure:
+
+```yaml
+telemetry:
+  logging:
+    level: info
+    encoding: console # console, json
+  # other configs
+
+api:
+  http:
+    listen_addr: 0.0.0.0:7685
+    max_body_size: "2Mi"
+    tls:
+      ca_path:
+      cert_path:
+    # other configs
+
+  grpc:
+    listen_addr: 0.0.0.0:7686
+    tls:
+      ca_path:
+      cert_path:
+    # other configs
+
+routes:
```

Comment:
@mkrueger12 our conversation around embeddings API in GEP0002 and ideas to support TTS & STT models have opened my eyes on the previously proposed strategy which is to have one uniform list of pools that support all API Glide providers.

I think that if you consider these two type of APIs, it's clear that the idea is not super viable. There is no reason to impose LLM lifecycle on something like STT/TTS. They are just too different. And we should take care of them considering their specific. 

Language, embeddings, transcribers, synthesizers are just too different to treat them the same way and we might end up being in trouble trying to go that route. 

So the new idea is to have type-specific model pools (e.g. language, embeddings, transcribers, synthesizers). This should aid separating logic needed (fallbacking, load balancing, unified request/response schemas, etc). for one type of models from logic that is appropriate in another. 

The current config examples have reflected this approach.

Feel free to take a look and let know how does that feel 🙌

Reply:
Yep this makes complete sense. Lets go that route